### PR TITLE
Refactor CLI-parsing

### LINKED ISSUE #6669: No `--version` argument
Raiden has a `version` subcommand, but no `--version` or `-v` flag. 

I think it's common to have a verison flag, and I'd argue we should change from the subcommand to the flag.

### LINKED ISSUE #6779: Monitoring not enabled by default in production mode
## Problem Definition

The `--enable-monitoring` option defaults to `False`.
On mainnet / production mode it needs to be enabled by default to provide a safe default for users and prevent possible loss of funds.

The flag should be changed to a binary choice (e.g. `--enable-monitoring` / `--no-enable-monitoring`).

We need to work out how to implement this with the current system.

### PR DESCRIPTION
**Pull request by ezdac**
## Description
Refactor the config-file option parsing, since before it was not re-usable and very fine-grained, replicating `click` library's internal behaviour.

What this Refactoring does in detail:

- Now CLI options can additionally have a `Parser`-classes concrete implementation-class attached, which can alter all other parsed command line options (for e.g. config-file parsing based on a config-file CLI option, for injecting/overwriting options based on a provided policy-option, etc.)
- Priorities can be provided to the additional parsers to determine option overwrite-precedence
- The parser-wrapped command functions (e.g. `run` / `smoketest`) have now better introspection into what source 
set a specific option, via the `Context.get_parameter_source( option_name )`, which returns a `click.core.ParameterSource` enum.

The ParameterSource contains the following values:
```
ParameterSource.COMMANDLINE - the option was specifically set by the command line by the user
ParameterSource.DEFAULT - the option was set by the default value, as specified in the `option( )` decorator
ParameterSource.DEFAULT_MAP - *an extra Parser set the option*, or the option's default was overwritten by click's `default_map`
ParameterSource.ENVIRONMENT - the option was set by an environment variable
ParameterSource.PROMPT - the option was set by a user prompt
```
(Read more about the `default_map` [here](https://click.palletsprojects.com/en/7.x/commands/#overriding-defaults))

This is easily possible, since click will add quite some convenience functionality in version 8.
Unfortunately this means using a release candidate for now.

- Introduce `click>=8.0.0a1` RC for its improved functionality
- Change modified `click` classes to comply to version `8.0.0a1` RC

Fixes #6669 
Fixes #6779

### CODE REVIEW

**Comment by ezdac** [comment]
The scenario-player is failing for this, since 5911b5a changed the `RaidenTestSetup.args['password_file']`
type.
In order for the checks to pass, the SP would have to be updated - but this would make SP runs fail 
during the time this PR is not merged.

I will wait for a final approval to push changes in the SP forward.

File: raiden/ui/cli.py
Comment (ulope) [inline raiden/ui/cli.py:150]: Nitpick this prints the internal option name.

**Review by ulope — COMMENTED** [review]
I like this very much. Nice work!
Some comments and questions below.

I did notice the following which seems unexpected:
- Put `rpc = False` in a config file
- Run Raiden with `--config-file <thefile> --api-address 0.0.0.0:1234`
- No warning is printed about RPC being needed for `--api-address`

**Comment by ezdac** [comment]
> * Put `rpc = False` in a config file
> * Run Raiden with `--config-file <thefile> --api-address 0.0.0.0:1234`
> * No warning is printed about RPC being needed for `--api-address`

Mhh, you are right.
This is due to the fact that we don't pipe an updated `default_map` through the whole click parsing logic again. That's what I tried at first - but failed to do - by recursively calling the `Group`- wrapped function.
So with this approach we will (somehow) always have this problem that the click parsing already happened as soon as we read in the e.g. config-file values.
What is done now is that I replicate the internals of click to some degree, but not the full degree in the `Group.invoke()`.
So this approach is a little bit more generalised and extendable then the approach we had before,
but still is similar.

Regarding the cause of the problem with the`--rpc` check that you noticed: 
The checking callback is registered on the `webui` option - which will not be processed again when your config file is read. Even if i replicate the callback invokation in our `Group.invoke()`, it will only be checked when the `--web-ui` is also present in the config file.

File: raiden/ui/cli.py
Comment (ezdac) [reply raiden/ui/cli.py]: The reasoning was that someone providing explicit config file / cli options knows what they are doing.
But I think you are right - changed this to always warn.

File: raiden/utils/cli.py
Comment (ezdac) [reply raiden/utils/cli.py]: No! :)

**Comment by ezdac** [comment]
RPC checking now is done in the `run( )` function and the problem you mentioned is resolved.

**Comment by ezdac** [comment]
There is a circular dependency between this and https://github.com/raiden-network/scenario-player/pull/657

Does someone object to merging without passing the `scenario-player-integration-test` check?
(And also, can someone with more privileges do so, if possible? :) )

**Review by ulope — APPROVED** [review]

**Pull request merged**
